cilium-health/cmd/root.go

```diff
@@ -131,7 +135,17 @@ func initConfig() {
 
 func runServer() {
 	common.RequireRootPrivilege(targetName)
-	os.Remove(defaults.SockPath)
+
+	// Write the pidfile (if specified)
+	if pidfile := viper.GetString("pidfile"); pidfile != "" {
+		pid := os.Getpid()
+		pidBytes := []byte(strconv.Itoa(pid) + "\n")
+		if err := ioutil.WriteFile(pidfile, pidBytes, 0660); err != nil {
+			errorString := fmt.Sprintf("Unable to write pidfile %s: %s", pidfile, err)
+			fmt.Println(errorString)
+			os.Exit(-1)
+		}
+	}
 
```

Comment:
Could you move this pidfile creation into a separate package? Would be nice to get rid of the `facebookgo/pidfile ` dependency I added to the agent.

Reply:
Sure.

Reply:
Done.